src/mongocxx/exception/private/error_category.cpp

```diff
@@ -382,7 +383,12 @@ class mongocxx_error_category_impl final : public std::error_category {
     }
 
     std::string message(int code) const noexcept override {
-        return "unknown mongocxx error";
+        switch (static_cast<error_code>(code)) {
+        case error_code::k_database_uninitialized:
+            return "database is uninitialized";
```

Comment:
Lets make this a little more detailed. 'database' is ambiguous, overall in our context. I think adding the classname here would be better: "mongocxx::database object used when not initialized". Same overall comment about 'initialized", but I don't immediately have a better suggestion